language/move-analyzer/src/completion.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::context::Context;
+use lsp_server::Request;
+use lsp_types::{CompletionItem, CompletionItemKind, CompletionParams, Position};
+use move_command_line_common::files::FileHash;
+use move_lang::{
+    diagnostics::Diagnostic,
+    parser::lexer::{Lexer, Tok},
+};
+use std::collections::HashSet;
+
+/// A static list of all possible keywords in the Move language, including contextual ones.
+const KEYWORDS: &[&str] = &[
+    "abort",
+    "aborts_if",
+    "aborts_with",
+    "acquires",
+    "address",
+    "apply",
+    "as",
+    "assume",
+    "axiom",
+    "break",
+    "choose",
+    "const",
+    "continue",
+    "decreases",
+    "else",
+    "emits",
+    "ensures",
+    "except",
+    "forall",
+    "friend",
+    "fun",
+    "global",
+    "has",
+    "if",
+    "include",
+    "internal",
+    "invariant",
+    "let",
+    "local",
+    "modifies",
+    "module",
+    "mut",
+    "min",
+    "native",
+    "post",
+    "phantom",
+    "pragma",
+    "public",
+    "requires",
+    "Self",
+    "schema",
+    "script",
+    "spec",
+    "struct",
+    "succeeds_if",
+    "to",
+    "update",
+    "use",
+    "with",
+];
+
+/// Constructs an `lsp_types::CompletionItem` with the given `label` and `kind`.
+fn completion_item(label: &str, kind: CompletionItemKind) -> CompletionItem {
+    CompletionItem {
+        label: label.to_owned(),
+        kind: Some(kind),
+        detail: None,
+        documentation: None,
+        deprecated: None,
+        preselect: None,
+        sort_text: None,
+        filter_text: None,
+        insert_text: None,
+        insert_text_format: None,
+        insert_text_mode: None,
+        text_edit: None,
+        additional_text_edits: None,
+        command: None,
+        commit_characters: None,
+        data: None,
+        tags: None,
```

Comment:
Nit: similar to above above possibly using struct update with `Default::default` here for these None fields
```suggestion
        label: label.to_owned(),
        kind: Some(kind),
       ..Default::default()
```